api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/AddressTest.java

```diff
@@ -0,0 +1,112 @@
+    private static final String LONG_STRING = CharBuffer.allocate(501).toString().replace('\0', 'A');
+
+    private Validator validator;
+
+    @BeforeEach
+    void setup() {
+        ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+        validator = validatorFactory.getValidator();
+    }
+
+    @Test
+    void shouldValidateAddressSuccessfully() {
+        //Given
+        var address = aValidAddress();
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasNoViolations();
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithNoAddressLine1() {
+        //Given
+        var address = anAddressWithAddressLine1(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("must not be null", "addressLine1");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWhenAddressLine1IsTooLong() {
+        //Given
+        var address = anAddressWithAddressLine1(LONG_STRING);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("size must be between 1 and 500", "addressLine1");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWhenAddressLine2IsTooLong() {
+        //Given
+        var address = anAddressWithAddressLine2(LONG_STRING);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("size must be between 0 and 500", "addressLine2");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithNoTownOrCity() {
+        //Given
+        var address = anAddressWithTownOrCity(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("must not be null", "townOrCity");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWhenTownOrCityIsTooLong() {
+        //Given
+        var address = anAddressWithTownOrCity(LONG_STRING);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("size must be between 1 and 500", "townOrCity");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithNoPostcode() {
+        //Given
+        var address = anAddressWithPostcode(null);
+        //When
+        Set<ConstraintViolation<Address>> violations = validator.validate(address);
+        //Then
+        assertThat(violations).hasSingleConstraintViolation("must not be null", "postcode");
+    }
+
+    @Test
+    void shouldFailToValidateAddressWithInvalidPostcode() {
```

Comment:
Can we have a parameterised test with lots of values here please rather than just a single one? Blank String, too long, too short, invalid characters etc etc.